assets/mdk/proof-dependencies.mdk

```diff
@@ -0,0 +1,87 @@
+Colorizer     : dafnyx
+
+Heading Depth: 0
+
+Css Header:
+  body.madoko {
+    max-width: 100%;
+    margin: 0;
+    padding: 0;
+  }
+  .page-content {
+    padding: 0;
+  }
+
+# Introduction
+
+A verified program is only as good as its specification. If the properties you want to prove about your program are contradictory, any program will satisfy them. If the properties you want to prove don't cover all of the program's behavior, any implementation will satisfy the uncovered portions. Specification problems of these sort result in verification that is technically sound, but that may provide a false sense of security that the program behaves as intended.
```

Comment:
I'm not fond of the saying "A verified program is only as good as its specification" because it places the bar unnecessarily high for verification to be valuable. If I verify that that my sorting algorithm and specify only that it doesn't have runtime errors, it doesn't mean that it's no good for sorting. I assume that what you mean is that specs, like programs, can have errors, but I don't know if that sentence captures that accurately.

Reply:
Likewise, I don't like to describe underspecification as a problem.

Reply:
Yeah, that's a good point. I'll reword that sentence.

And I also don't like treating underspecification as a problem, either. I'll try to avoid implying that. It's only a problem if you've specified less than you intended to.